drivers/volume/portworx/portworx.go

```diff
@@ -1518,7 +1508,17 @@ func (p *portworx) generateStatusReponseFromTaskIDs(taskIDs []string, credID str
 
 		var conditions []crdv1.VolumeSnapshotCondition
 
-		if csStatus.status == api.CloudBackupStatusDone {
+		if isCloudsnapStatusFailed(csStatus.status) {
+			logrus.Errorf(csStatus.msg)
```

Comment:
Pass in the groupsnapshot object to `generateStatusReponseFromTaskIDs()` and log with `GroupSnapshotLog()` so that there is more context in the log.

Reply:
done